packages/github/src/github.ts

```diff
@@ -1,158 +1,36 @@
-    ]
-  }
-
-  private static getOctokitOptions(
-    args: [string, Octokit.Options]
-  ): Octokit.Options {
-    const token = args[0]
-    const options = {...args[1]} // Shallow clone - don't mutate the object provided by the caller
-
-    // Base URL - GHES or Dotcom
-    options.baseUrl = options.baseUrl || this.getApiBaseUrl()
-
-    // Auth
-    const auth = GitHub.getAuthString(token, options)
-    if (auth) {
-      options.auth = auth
-    }
-
-    // Proxy
-    const agent = GitHub.getProxyAgent(options.baseUrl, options)
-    if (agent) {
-      // Shallow clone - don't mutate the object provided by the caller
-      options.request = options.request ? {...options.request} : {}
-
-      // Set the agent
-      options.request.agent = agent
-    }
-
-    return options
+const baseUrl = Utils.getApiBaseUrl()
+const defaults = {
+  baseUrl,
+  request: {
+    agent: Utils.getProxyAgent(baseUrl)
   }
+}
 
-  private static getGraphQL(args: [string, Octokit.Options]): GraphQL {
-    const defaults: GraphQLRequestParameters = {}
-    defaults.baseUrl = this.getGraphQLBaseUrl()
-    const token = args[0]
-    const options = args[1]
-
-    // Authorization
-    const auth = this.getAuthString(token, options)
-    if (auth) {
-      defaults.headers = {
-        authorization: auth
-      }
-    }
-
-    // Proxy
-    const agent = GitHub.getProxyAgent(defaults.baseUrl, options)
-    if (agent) {
-      defaults.request = {agent}
-    }
-
-    return graphql.defaults(defaults)
-  }
-
-  private static getAuthString(
-    token: string,
-    options: Octokit.Options
-  ): string | undefined {
-    // Validate args
-    if (!token && !options.auth) {
-      throw new Error('Parameter token or opts.auth is required')
-    } else if (token && options.auth) {
-      throw new Error(
-        'Parameters token and opts.auth may not both be specified'
-      )
-    }
-
-    return typeof options.auth === 'string' ? options.auth : `token ${token}`
-  }
-
-  private static getProxyAgent(
-    destinationUrl: string,
-    options: Octokit.Options
-  ): http.Agent | undefined {
-    if (!options.request?.agent) {
-      if (httpClient.getProxyUrl(destinationUrl)) {
-        const hc = new httpClient.HttpClient()
-        return hc.getAgent(destinationUrl)
-      }
-    }
-
-    return undefined
-  }
-
-  private static getApiBaseUrl(): string {
-    return process.env['GITHUB_API_URL'] || 'https://api.github.com'
-  }
-
-  private static getGraphQLBaseUrl(): string {
-    let url =
-      process.env['GITHUB_GRAPHQL_URL'] || 'https://api.github.com/graphql'
-
-    // Shouldn't be a trailing slash, but remove if so
-    if (url.endsWith('/')) {
-      url = url.substr(0, url.length - 1)
-    }
-
-    // Remove trailing "/graphql"
-    if (url.toUpperCase().endsWith('/GRAPHQL')) {
-      url = url.substr(0, url.length - '/graphql'.length)
-    }
-    return url
-  }
+export const GitHub = Octokit.plugin(
```

Comment:
Plugins make this really simple. @octokit/rest, which is what we are using in production now, only implements 3 plugins, the two listed here and an additional logging plugin. [source](https://github.com/octokit/rest.js/blob/master/src/index.ts)

We want to save package size, so we don't implement the logging plugin as its default log level doesn't even log, but the `paginateRest` adds the paginate function which is helpful, and `restEndpointMethods` adds nearly all of the rest calls (stuff like `octokit.repos.getBranch`. These two plugins brings us to feature parity with `@actions/github` v2, except for the registerEndpoint method which was deprecated in the newest version of @octokit/rest, and a few other legacy apis.

Default values just set the defaults for a certain field unless a user overrides it. We use that for proxy settings and